src/index.ts

```diff
@@ -22,264 +19,303 @@ export interface ResponseBody extends Promise<Response> {
   void(): Promise<void>
 }
 
-export interface Options<P extends Payload> extends RequestInit {
-  /** Resource URL */
+interface ResponsePromise<P extends Payload = Payload>
+  extends Promise<Response<P>>,
+    BodyMethods {}
+
+interface Serializer {
+  (params: SearchParams): URLSearchParams | string
+}
+
+interface RequestMethod<P extends Payload> {
+  <T extends P>(
+    resource?: number | string | RequestMethodOptions<T>,
+    options?: RequestMethodOptions<T>
+  ): ResponsePromise<T>
+}
+
+interface RequiredOptions<P extends Payload> extends RequestInit {
+  /**
+   * Part of the request URL
+   */
   resource?: string
```

Comment:
it's a bit funny to see optional params inside something called "RequiredOptions" 😅

Reply:
It's internal :) 
Required for the request.